src/framework/theme/components/chat/custom-message.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable, TemplateRef } from '@angular/core';
+
+/**
+ * Service for self register custom templates
+ */
+@Injectable()
+export class NbCustomMessageService {
+
+  protected readonly customMessages = new Map<string, TemplateRef<any>>();
+
+  getMessageTemplate(type: string): TemplateRef<any> | undefined {
+    return this.customMessages.get(type);
+  }
+
+  setMessageTemplate(type: string, template: TemplateRef<any>): void {
+    this.customMessages.set(type, template);
+  }
+
+  delete(type: string): boolean {
```

Comment:
Other methods have a `MessageTemplate` in the names, shouldn't this method also has it? Also, if we go with `register...` above, let's change it to `unregister...`.